### ISSUE TITLE
Can not use java.util.logging in my data class?

### ISSUE THREAD

**Issue by dapengzhang0** [JetBrains/lincheck#116]
I want to use java.util.logging to print debugging information. But the Lincheck test would fail if I add java logging in my data class.

```
> linCheck FAILED
    java.lang.InternalError: CallerSensitive annotation expected at frame 1
        at java.base/jdk.internal.reflect.Reflection.getCallerClass(Native Method)
        at org.jetbrains.kotlinx.lincheck.tran$f*rmed.java.util.logging.Logger.getLogger(Logger.java:701)
```

**Comment by ndkoval** [comment]
Unfortunately, I need to see the failing code to help you. I am also not sure that you really want to print debugging information this way, as Lincheck executes your methods hundreds of thousands of times.

**Comment by dapengzhang0** [comment]
M code is here

https://github.com/grpc/grpc-java/blob/0e64316f1d010ff9e547d3fbab4d4788a58bbe8a/servlet/src/main/java/io/grpc/servlet/AsyncServletOutputStreamWriter.java

I've already worked around this issue by abstracting java.util.logging.Logger class into my `Log` interface, and test with a noop `Log` instance.

https://github.com/grpc/grpc-java/blob/0e64316f1d010ff9e547d3fbab4d4788a58bbe8a/servlet/src/test/java/io/grpc/servlet/AsyncServletOutputStreamWriterConcurrencyTest.java

Then the lincheck worked. However, abstracting Log interface just for lincheck made the data class kind of hacky. I hope lincheck could just load the class successfully and ignore all interleaving with java.util.logging.Logger when `addGuarantee()`.

**Comment by dapengzhang0** [comment]
>  I am also not sure that you really want to print debugging information this way, as Lincheck executes your methods hundreds of thousands of times.

These are FINE or FINEST level logs, so it won't really print anything in the test because the logging level is not at FINE in the test.

**Comment by alefedor** [comment]
@dapengzhang0

**Comment by alefedor** [comment]
@dapengzhang0 you are right, it seems like an option not to transform a class is useful, especially for such loggers, which don't affect correctness. 

Yet, this issue arises only for `java.util` classes, so maybe we'll just add a corner case for `java.util.logging` in the next release

**Comment by ndkoval** [comment]
@alefedor I would suggest adding an option to load the specified classes (specified via regex or so) via the system class loader. We already have an issue with Kover caused by its classes transformation.

**Comment by ndkoval** [comment]
@alefedor could you please provide a fix when you have time?

**Comment by ndkoval** [comment]
This issue will likely be fixed automatically under #136

**Comment by ndkoval** [comment]
@dapengzhang0, I'm closing this issue, as it should be fixed now. Please reopen it if you are still facing the bug.
